source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -455,18 +469,33 @@ I/O.
 Checking Out a Connection
 -------------------------
 
-A Pool MUST have a method of allowing the driver to check out a `Connection <#connection>`_. Checking out a `Connection <#connection>`_ involves entering the WaitQueue and waiting for a `Connection <#connection>`_ to become available. If the thread times out in the WaitQueue, an error is thrown.
-
-If, in the process of iterating available `Connections <#connection>`_ in the
-pool by the checkOut method, a perished `Connection <#connection>`_ is
-encountered, such a `Connection <#connection>`_ MUST be closed (as described in
-`Closing a Connection <#closing-a-connection-internal-implementation>`_) and the
-iteration of available `Connections <#connection>`_ MUST continue until either a
-non-perished available `Connection <#connection>`_ is found or the list of
-available `Connections <#connection>`_ is exhausted. If no `Connections
-<#connection>`_ are available and the total number of `Connections
-<#connection>`_ is less than maxPoolSize, the pool MUST create a `Connection
-<#connection>`_, establish it, mark it as "in use" and return it.
+A Pool MUST have a method of allowing the driver to check out a `Connection`_.
+Checking out a `Connection`_ involves entering the WaitQueue, waiting to be
+granted access to the list of available connections, and finding or creating a
+`Connection`_ to be returned. If the thread times out in the WaitQueue, an error
+is thrown.
+
+Once reaching the front of the WaitQueue, a thread begins iterating over the
```

Comment:
Discussion in meeting, we should replace "thread" with "request":
> Once reaching the front of the WaitQueue, a request begins iterating over the

Reply:
I opted for slightly different wording (i.e. "the pool begins iterating...") to avoid having the request do any work / seem like a thread. Let me know what you think.